backend/src/app/services/site/site.service.ts

```diff
@@ -309,28 +313,50 @@ export class SiteService {
    * @param siteId site Id
    * @returns a single site matching the site ID
    */
-  async findSiteBySiteId(siteId: string, pending: boolean) {
+  async findSiteBySiteId(siteId: string, pending: boolean, userInfo: any) {
     this.sitesLogger.log('SiteService.findSiteBySiteId() start');
     this.sitesLogger.debug('SiteService.findSiteBySiteId() start');
     const response = new FetchSiteDetail();
 
     response.httpStatusCode = 200;
 
-    if (pending) {
-      const result = await this.siteRepository.findOne({
-        where: { id: siteId, srAction: SRApprovalStatusEnum.PENDING },
-      });
+    let snapShot: Snapshots = null;
+
+    if (!userInfo) {
+      this.sitesLogger.log('SiteService.findSiteBySiteId() user not logged in');
 
-      response.data = result ? result : null;
+      snapShot = null;
+    } else if (userInfo?.identity_provider === 'idir') {
+      this.sitesLogger.log(
+        'SiteService.findSiteBySiteId() idir user - no snapshot',
+      );
     } else {
-      const result = await this.siteRepository.findOne({
-        where: { id: siteId },
-      });
+      snapShot = await this.snapShotService.getMostRecentSnapshot(
+        siteId,
+        userInfo.sub,
+      );
+    }
 
-      response.data = result ? result : null;
+    if (!snapShot) {
+      if (pending) {
+        const result = await this.siteRepository.findOne({
+          where: { id: siteId, srAction: SRApprovalStatusEnum.PENDING },
+        });
+
+        response.data = result ? result : null;
+      } else {
+        const result = await this.siteRepository.findOne({
+          where: { id: siteId },
+        });
+
+        response.data = result ? result : null;
+      }
+      this.sitesLogger.log('SiteService.findSiteBySiteId() end');
+      this.sitesLogger.debug('SiteService.findSiteBySiteId() end');
```

Comment:
you can have the logging after the if-else block

Reply:
done